### PR TITLE
Allow ability to delete IAM Policies

### PR DESCRIPTION
**Pull request by cpollard0**
Add support for deleting IAM policies: 

```
  - name: iam-policy-unused
    resource: iam-policy
    filters:
      - type: unused
      - type: value
        key: "Arn"
        op: regex
        value: "(arn:aws:iam::[0-9])"
        # regular expression to filter out AWS managed policies. 
        # aws policies in the format arn:aws:iam::aws whereas our policies are in the format arn:aws:iam::${account_number}
    actions:
      - delete
```

### CODE REVIEW

**Comment by kapilt** [comment]
looks like some whitespace lint issues, (`make lint` for local check)

**Comment by cpollard0** [comment]
Thanks for the tip @kapilt!

File: c7n/resources/iam.py [hunk 1]
```diff
@@ -536,6 +536,43 @@ def process(self, resources, event=None):
...
+            - delete
+
+    """
+    schema = type_schema('delete')
+    permissions = ('iam:DeletePolicy',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('iam')
+        for r in resources:
+            self.log.debug('Deleting policy %s' % r['PolicyName'])
```
Comment (kapilt) [inline c7n/resources/iam.py]: lgtm, but should we be filtering out managed policies in the action as a safety guard?

Comment (cpollard0) [reply c7n/resources/iam.py]: That's a good point; you can't delete managed policies. I'll update the PR to reflect this.

**Comment by cpollard0** [comment]
@kapilt  - How does this look?

**Pull request merged**